internal/resource/config/sp/adapters/sp_adapter_resource_gen.go

```diff
@@ -0,0 +1,428 @@
+}
+
+type spAdapterResourceModel struct {
+	AdapterId             types.String `tfsdk:"adapter_id"`
+	AttributeContract     types.Object `tfsdk:"attribute_contract"`
+	Configuration         types.Object `tfsdk:"configuration"`
+	Name                  types.String `tfsdk:"name"`
+	ParentRef             types.Object `tfsdk:"parent_ref"`
+	PluginDescriptorRef   types.Object `tfsdk:"plugin_descriptor_ref"`
+	TargetApplicationInfo types.Object `tfsdk:"target_application_info"`
+}
+
+func (r *spAdapterResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"adapter_id": schema.StringAttribute{
+				Required:    true,
+				Description: "The ID of the plugin instance. The ID cannot be modified once the instance is created.<br>Note: Ignored when specifying a connection's adapter override.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"attribute_contract": schema.SingleNestedAttribute{
+				Attributes: map[string]schema.Attribute{
+					"core_attributes": schema.ListNestedAttribute{
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"name": schema.StringAttribute{
+									Required:    true,
+									Description: "The name of this attribute.",
+								},
+							},
+						},
+						Optional:    false,
+						Computed:    true,
+						Default:     listdefault.StaticValue(coreAttributesDefault),
+						Description: "A list of read-only attributes that are automatically populated by the SP adapter descriptor.",
+					},
+					"extended_attributes": schema.ListNestedAttribute{
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"name": schema.StringAttribute{
+									Required:    true,
+									Description: "The name of this attribute.",
+								},
+							},
+						},
+						Optional:    true,
+						Computed:    true,
+						Default:     listdefault.StaticValue(extendedAttributesDefault),
+						Description: "A list of additional attributes that can be returned by the SP adapter. The extended attributes are only used if the adapter supports them.",
+					},
+				},
+				Optional:    true,
+				Computed:    true,
+				Default:     objectdefault.StaticValue(attributeContractDefault),
+				Description: "A set of attributes exposed by an SP adapter.",
+			},
+			"configuration": pluginconfiguration.ToSchema(),
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The plugin instance name. The name can be modified once the instance is created.<br>Note: Ignored when specifying a connection's adapter override.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"parent_ref": schema.SingleNestedAttribute{
```

Comment:
Should we remove these parent_ref attrs from resources in the provider? I'm not sure they make sense in a terraform context.

Reply:
Good question!

Reply:
We'll leave them in - I can see a benefit here, because you can opt to leave all the fields out of the configuration and they will still inherit from the parent